### PR TITLE
[1LP][RFR]Workaround for MoveTargetOutOfBoundsException

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
Workaround for navigation to avoid MoveTargetOutOfBoundsException
{{pytest: -k 'test_add_cancelled_validation or test_hostname_max_character_validation or test_host_name_max_character_validation or test_empty_discovery_form_validation or test_keypair_add_and_remove_tag' -vvv --long-running --use-provider complete --use-template-cache}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 42445074726d40e23140cfdd179940dd2f231577:

`cfme/services/catalogs/catalog_item.py`:
- :red_circle: Line 57:11: **N801** *class names should use CapWords convention*
- :bangbang: Line 57:27: **E262** *inline comment should start with '# '*
- :red_circle: Line 65:11: **N801** *class names should use CapWords convention*
- :bangbang: Line 65:28: **E262** *inline comment should start with '# '*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fe350097ac514ed454399f9905b71eab09c213e8:

`cfme/cloud/keypairs.py`:
- :heavy_exclamation_mark: Line 242:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/infrastructure/provider/__init__.py [hunk 1]
```diff
@@ -217,8 +217,13 @@ class Add(CFMENavigateStep):
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        self.prerequisite_view.toolbar.configuration.item_select('Add a New '
-                                                                 'Infrastructure Provider')
+        try:
```
[the remark this answers is not in the document]
Comment (rlbabyuk) [reply cfme/infrastructure/provider/__init__.py]: This is a workaround for navigation as in some places as for bz 1475303, when flash message appears we need to click twice on toolbar to select option

**Comment by mshriver** [comment]
Acceptable workaround for BZ

**Comment by mshriver** [comment]
PRT still running, but keypair test passed before quota filled.

**Pull request merged**